lambdas/service/app.py

```diff
@@ -1524,6 +1528,26 @@ def handle_manifest_generation_request() -> Tuple[int, Manifest]:
                     catalog=IndexName.validate_catalog_name,
                     filters=str,
                     token=str)
+
+    def amend_manifest_location(manifest: Manifest, token: str = None):
+        url = furl(app.self_url())
+        if 'fetch' in url.path.segments:
+            url.path.segments.remove('fetch')
+            if token is None:
+                url.args.update({'catalog': catalog,
+                                 'filters': json.dumps(filters),
+                                 'format': format_.value})
+            else:
+                token = async_service.decode_token(token)
+                url.args.update({'catalog': token['catalog'],
+                                 'filters': token['filters'],
+                                 'format': token['format']})
+            generator = ManifestGenerator.for_format(format_, service, catalog, filters)
+            manifest = Manifest(location=url.url,
+                                was_cached=manifest.was_cached,
+                                properties=generator.manifest_properties(url.url))
+        return 0, manifest
```

Comment:
Either use a local variable so it's clear what the zero means in this context:

```suggestion
        retry_after = 0
        return retry_after, manifest
```
Or, preferably IMO, just drop it:
 
```suggestion
        return manifest
```